.github/workflows/test.yml

```
@@ -0,0 +1,61 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - "ubuntu-latest"
+          - "macos-latest"
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: make build
+      - name: Test
+        run: |
+          go install gotest.tools/gotestsum@latest
+          gotestsum --format testname --junitfile ${{ matrix.platform }}-junit-report.xml
```

What do you think if those steps are added to the `test` Make goal?

```suggestion
        run: make test TEST_REPORT=${{ matrix.platform }}-junit-report.xml
```

---

https://github.com/elastic/apm-aws-lambda/pull/362/commits/37f2b84fc08fc9e4dacb7d8fdc6afe114cc41288